core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchangeResource.java

```diff
@@ -115,6 +128,11 @@ public static String getTokenUri(UUID authId)
         return TOKEN_ENDPOINT + authId;
     }
 
+    public static String getInitiateUri(UUID authId)
+    {
+        return TOKEN_ENDPOINT + "initiate/" + hashAuthId(authId);
```

Comment:
What is the point of hashing UUID? UUID is so short that could be simply sent as parameter.

Reply:
The hash protects the UUID from the browser.  The browser no longer gets the UUID at all and only ever sees the hash, so a compromised browser can not steal an auth token.  This is the same trick that nonce tokens uses.